go-controller/pkg/node/ovspinning/ovspinning_linux.go

```diff
@@ -182,11 +182,17 @@ func setProcessCPUAffinity(targetPIDStr string) error {
 		return nil
 	}
 
-	klog.Infof("Setting CPU affinity of PID(%d) to %s, was %s", targetPID, printCPUSet(currentProcessCPUs), printCPUSet(targetProcessCPUs))
-
-	err = unix.SchedSetaffinity(targetPID, &currentProcessCPUs)
+	taskIDs, err := getThreadsOfProcess(targetPID)
 	if err != nil {
-		return fmt.Errorf("can't set CPU affinity of PID(%d) to %s: %w", targetPID, printCPUSet(currentProcessCPUs), err)
+		return fmt.Errorf("can't get tasks of PID(%d):%w", targetPID, err)
+	}
+
+	klog.Infof("Setting CPU affinity of PID(%d) (ntasks=%d) to %s, was %s", targetPID, len(taskIDs), printCPUSet(currentProcessCPUs), printCPUSet(targetProcessCPUs))
+	for _, taskID := range taskIDs {
+		err = unix.SchedSetaffinity(taskID, &currentProcessCPUs)
+		if err != nil {
+			return fmt.Errorf("can't set CPU affinity of task(%d) PID(%d) to %s: %w", taskID, targetPID, printCPUSet(currentProcessCPUs), err)
```

Comment:
This needs to be a bit resilient. Imagine OVS is reconfiguring its threads at this very time and one of the threads disappears while you are in the loop. The error will end the processing and the rest of the threads will not be configured.

Maybe that is not an issue since atm OVS always recreates all its threads, but it is still a hidden race.

Reply:
good point! going to fix this race by not breaking the loop.